placement/placement.go

```diff
@@ -110,6 +111,12 @@ func (ps snapshot) Validate() error {
 	if expectedTotal != actualTotal {
 		return errTotalShardsMismatch
 	}
+
+	for _, c := range shardCountMap {
+		if ps.rf != c {
+			return errInvalidShardsCount
```

Comment:
Perhaps provide a more detailed error message, e.g., `fmt.Errorf("invalid shard count for shard %d: expected %d, actual %d", shard, ps.rf, c)`